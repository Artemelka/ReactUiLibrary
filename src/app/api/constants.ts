const NAMESPACE_API = 'api/v1';
const LOCALIZATION_SERVICE = 'localization';

export const API_URL = {
    GET_ALL_DICTIONARY: `/${NAMESPACE_API}/${LOCALIZATION_SERVICE}/dictionary`,
    GET_LABELS: `/${NAMESPACE_API}/${LOCALIZATION_SERVICE}/labels`,
    GET_LOCALES: `/${NAMESPACE_API}/${LOCALIZATION_SERVICE}/locales`
};
