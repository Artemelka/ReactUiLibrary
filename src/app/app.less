.App {
    bottom: 0;
    //font-family: 'Fira Sans', sans-serif;
    left: 0;
    overflow: auto;
    position: fixed;
    right: 0;
    top: 0;
}
