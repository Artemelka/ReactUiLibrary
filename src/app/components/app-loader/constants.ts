import { nameSpaceCreator } from 'services';
import { StoreKeys } from './types';

export const APP_LOADER_REDUCER_KEY = 'appLoader';
export const initialState = {
    [StoreKeys.IS_LOADING]: true
};

export const APP_LOADER_ACTIONS = nameSpaceCreator(`@@${APP_LOADER_REDUCER_KEY}`)([
    'CHANGE_STATE'
]);
