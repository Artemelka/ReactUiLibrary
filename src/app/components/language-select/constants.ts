import { SelectOptions } from '../../elements/inputs/select/types';
import { NavigatorLanguage } from '../../../services/translate';

export const SELECT_WIDTH = 150;
export const selectOptions: Array<SelectOptions> = [
    {
        value: NavigatorLanguage.RU,
        title: 'russian-language'
    }, {
        value: NavigatorLanguage.EN,
        title: 'english-language'
    }
];
