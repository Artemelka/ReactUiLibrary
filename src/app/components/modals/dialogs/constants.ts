export const ModalDialogName = {
    FIRST: 'firstModal',
    SECOND: 'secondModal',
    THIRD: 'thirdModal'
};
