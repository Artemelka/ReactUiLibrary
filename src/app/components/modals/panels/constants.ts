export const ModalPanelName = {
    FIRST: 'firstPanel',
    SECOND: 'secondPanel',
    THIRD: 'thirdPanel'
};
