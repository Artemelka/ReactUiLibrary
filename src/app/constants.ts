// ROOT ID
export const APP_ROOT = 'app-root';
export const MODAL_ROOT = 'modal-root';
export const NOTIFICATION_ROOT = 'notification-root';

// ROOT CLASS NAME
export const PAGE_MODAL_CONTAINER_CLASS_NAME = 'Page__modal-container';
export const PAGE_NOTIFICATION_CONTAINER_CLASS_NAME = 'Page__notification-container';
export const TAG_NAME_DIV = 'div';

// URL
export const PROJECT_LINK = 'https://github.com/Artemelka/ReactUiLibrary';
export const HOME_URL = '/';
export const LIBRARY_URL = '/library';
export const INPUT_SUB_URL = '/inputs';
