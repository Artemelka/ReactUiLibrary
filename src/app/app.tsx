import * as React from 'react';
import { Route } from 'react-router';
import { TestHomePage, LibraryPage } from './pages';

import './app.less';

export class App extends React.Component<{}, {}> {
    render() {
        return (
            <div className="App">
                <Route path="/" exact component={TestHomePage} />
                {/*<Route path="/" exact component={LibraryPage} />*/}
                {/*<Route path="/library" component={LibraryPage} />*/}
            </div>
        );
    }
}

