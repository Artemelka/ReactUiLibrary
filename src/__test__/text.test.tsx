import React from 'react';
import { shallow, mount } from 'enzyme';
import { Text } from '../app/elements';
import { ComponentType, HeadingType } from '../app/elements/text/constants';

const TEST_TEXT = 'test text';

describe('Test TextParagraph element', () => {
    test('Rendering snapshot H1', () => {
        const output = shallow(<Text type={ComponentType.HEADING} headingType={HeadingType.H1}>{TEST_TEXT}</Text>);

        expect(output.dive()).toMatchSnapshot();
    });
    test('Rendering snapshot H2', () => {
        const output = shallow(<Text type={ComponentType.HEADING} headingType={HeadingType.H2}>{TEST_TEXT}</Text>);

        expect(output.dive()).toMatchSnapshot();
    });
    test('Rendering snapshot H3', () => {
        const output = shallow(<Text type={ComponentType.HEADING} headingType={HeadingType.H3}>{TEST_TEXT}</Text>);

        expect(output.dive()).toMatchSnapshot();
    });
    test('Rendering snapshot H4', () => {
        const output = shallow(<Text type={ComponentType.HEADING} headingType={HeadingType.H4}>{TEST_TEXT}</Text>);

        expect(output.dive()).toMatchSnapshot();
    });
    test('Rendering snapshot H5', () => {
        const output = shallow(<Text type={ComponentType.HEADING} headingType={HeadingType.H5}>{TEST_TEXT}</Text>);

        expect(output.dive()).toMatchSnapshot();
    });
    test('Rendering snapshot H6', () => {
        const output = shallow(<Text type={ComponentType.HEADING} headingType={HeadingType.H6}>{TEST_TEXT}</Text>);

        expect(output.dive()).toMatchSnapshot();
    });
    test('Rendering snapshot Paragraph', () => {
        const output = shallow(<Text type={ComponentType.PARAGRAPH}>{TEST_TEXT}</Text>);

        expect(output.dive()).toMatchSnapshot();
    });
    test('Rendering snapshot Span', () => {
        const output = shallow(<Text type={ComponentType.SPAN}>{TEST_TEXT}</Text>);

        expect(output.dive()).toMatchSnapshot();
    });
    test('Expect className modifier --upper', () => {
        const output = shallow(<Text upper type={ComponentType.PARAGRAPH}>{TEST_TEXT}</Text>);

        expect(output.dive().hasClass('Text--upper')).toBe(true);
    });
    test('Expect className modifier --bold', () => {
        const output = shallow(<Text bold type={ComponentType.PARAGRAPH}>{TEST_TEXT}</Text>);

        expect(output.dive().hasClass('Text--bold')).toBe(true);
    });
    test('Expect className modifier --light', () => {
        const output = shallow(<Text light type={ComponentType.PARAGRAPH}>{TEST_TEXT}</Text>);

        expect(output.dive().hasClass('Text--light')).toBe(true);
    });
    test('Expect console.warn with bold and light props', () => {
        const testMock = jest.fn();
        console.warn = testMock;
        const output = shallow(<Text light bold type={ComponentType.PARAGRAPH}>{TEST_TEXT}</Text>);

        expect(testMock).toHaveBeenCalled();

    });
    test('Expect light modifier with bold and light props', () => {
        const output = shallow(<Text light bold type={ComponentType.PARAGRAPH}>{TEST_TEXT}</Text>);

        expect(output.dive().hasClass('Text--light')).toBe(true);
    });
    test('Expect type paragraph without prop type', () => {
        const output = shallow(<Text>{TEST_TEXT}</Text>);

        expect(output.dive().hasClass('Text--paragraph')).toBe(true);
    });
});
