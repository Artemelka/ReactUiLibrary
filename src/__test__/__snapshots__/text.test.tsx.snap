// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TextParagraph element Rendering snapshot H1 1`] = `
<h1
  className="Text Text--heading-h1 Text--heading"
  style={Object {}}
>
  test text
</h1>
`;

exports[`Test TextParagraph element Rendering snapshot H2 1`] = `
<h2
  className="Text Text--heading-h2 Text--heading"
  style={Object {}}
>
  test text
</h2>
`;

exports[`Test TextParagraph element Rendering snapshot H3 1`] = `
<h3
  className="Text Text--heading-h3 Text--heading"
  style={Object {}}
>
  test text
</h3>
`;

exports[`Test TextParagraph element Rendering snapshot H4 1`] = `
<h4
  className="Text Text--heading-h4 Text--heading"
  style={Object {}}
>
  test text
</h4>
`;

exports[`Test TextParagraph element Rendering snapshot H5 1`] = `
<h5
  className="Text Text--heading-h5 Text--heading"
  style={Object {}}
>
  test text
</h5>
`;

exports[`Test TextParagraph element Rendering snapshot H6 1`] = `
<h6
  className="Text Text--heading-h6 Text--heading"
  style={Object {}}
>
  test text
</h6>
`;

exports[`Test TextParagraph element Rendering snapshot Paragraph 1`] = `
<p
  className="Text Text--paragraph"
  style={Object {}}
>
  test text
</p>
`;

exports[`Test TextParagraph element Rendering snapshot Span 1`] = `
<span
  className="Text Text--span"
  style={Object {}}
>
  test text
</span>
`;
